src/parser/src/Type.cpp

```diff
@@ -0,0 +1,22 @@
+#include "Type.hpp"
+
+/**
+ * Returns the amount of bits required to store a single element of the
+ * primitive type within CuMat
+ * @return
+ */
+int Typing::Type::offset() {
+    switch (primType) {
+        case PRIMITIVE::STRING:
+        case PRIMITIVE::BOOL:
+            return 8;
+            break;
+        case PRIMITIVE::INT:
+        case PRIMITIVE::FLOAT:
+            return 64;
+            break;
+        default:
+            return 8;
```

Comment:
I'd recommend not adding a default case when switching over an enum. Prevents a wrong default where we to add another type to the enum, as without we get a compile error (desirable).